File: Week 5 - Classes/Lab 5/Store/Store.py
from Product import Product
from CashRegister import CashRegister

class Store:

    def __init__(self):
        self.product = Product()
        self.cash_in_cash_register = CashRegister()
    
    def read_options(self):
        choice = input("Choice (s/r/v/c/x): ")
        return choice
        
    def use(self):
        options = self.read_options()
        while (options != 'x'):
            if options == 's':
                self.sell()
            elif options == 'r':
                self.restock()
            elif options == 'v':
                self.view_stock()
            elif options == 'c':
                self.view_cash()
            else:
                self.menu()
            options = self.read_options()
        print("Done")

    def read_number(self):
        number = int(input("Number: "))
        return number

    def sell(self):
        amount = self.read_number()
        if (self.product.has(amount)):
            print("Not enough stock") 
        else:
            self.product.sell(amount)
            self.cash_in_cash_register.update_cash(amount)
    
    def restock(self):
        self.product.restock(self.read_number())
    
    def view_stock(self):
        print(self.product.toString())

    def view_cash(self):
        print(self.cash_in_cash_register.toString())
    
    def menu(self):
        print("Menu options")
        print("s = sell")
        print("r = restock")
        print("v = view stock")
        print("c = view cash")
        print("x = exit")
    
if __name__ == "__main__":
    store = Store()
    store.use()

File: Week 5 - Classes/Lab 5/Store/Store.java
/**
 * The store has and sells one product: Sticky tape.
 *
 * You can sell and restock a product, view stock and view cash.
 */
public class Store {

    // insert main method here
    public static void main(String[] args) {
        new Store().use();  
    }

    // insert 2 fields here
    private Product product;
    private CashRegister cashInCashRegister;


    // insert 1 constructor here
    public Store() {
        product = new Product();
        cashInCashRegister = new CashRegister();
    }

    private char readOptions() {
        System.out.print("Choice (s/r/v/c/x): ");
        return In.nextChar();
    }

    public void use() {
        char options;
        while ((options = readOptions()) != 'x') {
            switch (options) {
                case 's':
                    sell();
                    break;
                case 'r':
                    restock();
                    break;
                case 'v':
                    viewStock();
                    break;
                case 'c':
                    viewCash();
                    break;
                default:
                    help();
            }
        }
        System.out.println("Done");
    }

    private double readNumber() {
        System.out.print("Number: ");
        return In.nextDouble();    
    }

    private void sell() {
        double amount = readNumber();
        if (product.has(amount)) {
            System.out.println("Not enough stock");
        }
        else {
            product.sell(amount);
            cashInCashRegister.updateCash(amount);
        }
    }
    
    private void restock() {
        product.restock(readNumber());
    }

    private void viewStock() {
        System.out.println(product);
    }

    private void viewCash() {
        System.out.println(cashInCashRegister);
    }

    private void help() {
        System.out.println("Menu options");
        System.out.println("s = sell");
        System.out.println("r = restock");
        System.out.println("v = view stock");
        System.out.println("c = view cash");
        System.out.println("x = exit");
    }
}
